applications/fishing-map/src/features/timebar/Timebar.tsx

```diff
@@ -128,30 +133,46 @@ const TimebarWrapper = () => {
 
   const activityCategory = useSelector(selectActivityCategory)
 
-  if (!start || !end) return null
+  const onMouseMove = useCallback(
+    (clientX: number, scale: (arg: number) => Date) => {
+      if (clientX === null) {
+        if (highlightedTime !== undefined) {
+          dispatch(disableHighlightedTime())
+        }
+      } else {
+        const start = scale(clientX - 10).toISOString()
+        const end = scale(clientX + 10).toISOString()
+        dispatch(setHighlightedTime({ start, end }))
+      }
+    },
+    [dispatch, highlightedTime]
+  )
 
-  const onMouseMove = (clientX: number, scale: (arg: number) => Date) => {
-    if (clientX === null) {
-      if (highlightedTime !== undefined) {
-        dispatch(disableHighlightedTime())
+  const [internalRange, setInternalRange] = useState<Range | null>(null)
+  const onChange = useCallback(
+    (e) => {
+      if (e.source === 'ZOOM_OUT_MOVE') {
+        setInternalRange({ ...e })
+        return
       }
-    } else {
-      const start = scale(clientX - 10).toISOString()
-      const end = scale(clientX + 10).toISOString()
-      dispatch(setHighlightedTime({ start, end }))
-    }
-  }
+      setInternalRange(null)
+      onTimebarChange(e.start, e.end)
+    },
+    [setInternalRange, onTimebarChange]
+  )
+
+  if (!start || !end) return null
 
   return (
     <div>
       <TimebarComponent
         enablePlayback={true}
         labels={labels}
-        start={start}
-        end={end}
+        start={internalRange ? internalRange.start : start}
```

Comment:
I was worried about having a render without the correct value when the internalRange is null and the start wasn't updated yet but seems to be working well!